x/mongo/driver/ocsp/ocsp.go

```diff
@@ -91,6 +104,43 @@ func Verify(ctx context.Context, connState tls.ConnectionState) error {
 	return nil
 }
 
+// getParsedResponse attempts to parse a response from the stapled OCSP data or by contacting OCSP responders if no
+// staple is present.
+func getParsedResponse(ctx context.Context, cfg config, connState tls.ConnectionState) (*Response, error) {
+	cachedResponse := cfg.cache.Get(cfg.ocspRequest)
+	stapledResponse, err := parseStaple(cfg, connState.OCSPResponse)
```

Comment:
Is there a reason that `parseStaple` doesn't return the result of `newResponse`?  It's reads oddly to me that you've got a "response" that you have to wash through "newResponse" for use.

Reply:
The main reason was that the concept of processing a response into a smaller subset wasn't introduced until this PR and I wanted to avoid making too many changes. I like the idea, though, so I made changed `parseStaple` to be called `processStaple` and return `ResponseDetails`. I also changed `contactResponders` to match.